### DISCUSSION TITLE
I wrote a CLI and don't see how to test it after examining the docs folder.

### DISCUSSION THREAD

**Discussion by dscotese** [avajs/ava#3121] [Q&A]
I use [prompt](https://github.com/heapwolf/prompt-sync) to gather keystrokes while my code runs events in the background.  For example, if the user just hits enter, a new task is added to an internal list to run a function called `report` which spits out some text to the console followed by a Date String and some letters or dots after the Date String.  If the user enters "risky" and it is successfully handled, then one of the letters after the Date String that follows the text spit out to the console changes from dot to R or from R to the dot.

I would like to write a test that checks that this character after the Date String switches between R and . each time the user enters risky.  Can Ava read what the code it tests spits out to the console, and if so, where is the documentation that explains how to test it?

**Comment by novemberborn** [comment]
AVA itself does not provide a mechanism, but you could probably instrument `process.stdout` to intercept.

A cleaner way of doing this would be to run your code in a child process so you have better control over STDIN and STDOUT.
